src/main/java/frc/robot/vision/sources/RobotPoseEstimatingVisionSource.java

```diff
@@ -0,0 +1,15 @@
+package frc.robot.vision.sources;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.vision.GyroAngleValues;
+import frc.robot.vision.rawdata.RawVisionAprilTagData;
+
+import java.util.Optional;
+
+public interface RobotPoseEstimatingVisionSource extends VisionSource<RawVisionAprilTagData> {
+
+	void updateGyroAngles(GyroAngleValues gyroAngleValues);
+
+	Optional<Rotation2d> getRobotHeading();
+
+}
```

Comment:
No, not all RobotPoseEstimatingVisionSources need to update their gyro angles, the multiVisionSources for robot pose should do the thing with instanceOf. And the "getRobotHeading" should not exist since the LimelightSource can just return its' pose with the botpos1 heading, and the PhotonSource should not have the heading seperated from the pose.
This whole class breaks SOLID